### DISCUSSION TITLE
New install issue

### DISCUSSION THREAD

**Discussion by cvclukes** [netbox-community/netbox#11468] [General]
I'm trying to install netbox for the first time and I've been going through the install documentation provided at https://docs.netbox.dev/en/stable/installation/3-netbox/ and I'm at the point of "Test the Application" and I'm getting the below error. At first it was that django was missing, so I added that, but now it says below and when I try to just browse to the URL or IP it says "This site can't be reached" refused to connect. 

/opt/netbox/netbox$ python3 manage.py runserver 0.0.0.0:8000 --insecure
usage: manage.py runserver [-h] [--ipv6] [--nothreading] [--noreload] [--version] [--settings SETTINGS] [--pythonpath PYTHONPATH] [--no-color] [--force-color] [--skip-checks] [addrport]
manage.py runserver: error: unrecognized arguments: --insecure

**Comment by bogdancordos** [comment]
Are you trying to run the test server from the virtual environment?
You need to activate your "venv" first (source /opt/netbox/venv/bin/activate) and then run the test server. You might have had installed Django on the global environment as well, which is not necessary if you have fallowed the installation steps.

**Reply by cvclukes** [reply]
I did not catch it needed to be in venv. Thank you, I've gotten it up and running now.

**Comment by pankajmamtani** [comment]
possible to migrate test server to production. 

My problem is it keeps going down then I have to run the below commands to get to up.

cd /opt/netbox
. venv/bin/activate
python3 netbox/manage.py runserver 0.0.0.0:8000 --insecure

**Reply by candlerb** [reply]
Production instances shouldn't use the development server.  You should run gunicorn under systemd, and have a reverse proxy like Apache or Nginx in front.

All of this is in the [installation documentation](https://docs.netbox.dev/en/stable/installation/) (steps 4 and 5).

**Reply by pankajmamtani** [reply]
ok thanks, my bad i missed it. Working further on the given steps, it given error - ln: failed to create symbolic link '/etc/nginx/sites-enabled/netbox': No such file or directory


>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
sudo cp /opt/netbox/contrib/nginx.conf /etc/nginx/sites-available/netbox
sudo rm /etc/nginx/sites-enabled/default
sudo ln -s /etc/nginx/sites-available/netbox /etc/nginx/sites-enabled/netbox
rm: cannot remove '/etc/nginx/sites-enabled/default': No such file or directory
ln: failed to create symbolic link '/etc/nginx/sites-enabled/netbox': No such file or directory
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

**Reply by candlerb** [reply]
You made a mistake. I would suggest in future you do the commands one at a time, so it's easier to see where things go wrong, rather than pasting all three at once.

```
rm: cannot remove '/etc/nginx/sites-enabled/default': No such file or directory
```

suggests to me that you didn't install the nginx package (see where it says "Begin by installing nginx:" in the instructions) - or perhaps you tried to, but there was an error and you carried on anyway, without sorting out the error.

EDIT: that doesn't explain why the "cp" didn't fail though.

What do these show?

```
ls -l /etc/nginx/sites-available
ls -l /etc/nginx/sites-enabled
```

**Reply by pankajmamtani** [reply]
please find the output

 ls -l /etc/nginx/sites-available
ls -l /etc/nginx/sites-enabled
total 0
drwxr-xr-x. 2 root root 24 Apr 21 10:57 netbox
total 0
lrwxrwxrwx. 1 root root 33 Apr 26 17:14 netbox -> /etc/nginx/sites-available/netbox


finally worked following the setups in this link since I am using rocky linux https://www.howtoforge.com/how-to-install-netbox-on-rocky-linux-9/

I used apache instead of nginx

**Reply by candlerb** [reply]
Please, one command at a time!!!

I will guess you say this:

```
# ls -l /etc/nginx/sites-available
total 0
drwxr-xr-x. 2 root root 24 Apr 21 10:57 netbox
```

That shows you have created a *directory* called netbox, instead of copying a file. You'll need to remove the contents of this directory, remove the directory, and copy the file again.

> I used apache instead of nginx

OK, then you can just remove the nginx package and all its config files.
